core/vm/ovm_state_manager.go

```diff
@@ -210,6 +210,20 @@ func testAndSetContractStorage(evm *EVM, contract *Contract, args map[string]int
 	return []interface{}{true}, nil
 }
 
+func hasEmptyAccount(evm *EVM, contract *Contract, args map[string]interface{}) ([]interface{}, error) {
+	address, ok := args["_address"].(common.Address)
+	if !ok {
+		return nil, errors.New("Could not parse address arg in hasEmptyAccount")
+	}
+
+	contractHash := evm.StateDB.GetCodeHash(address)
```

Comment:
Is it possible to have the nonce be 0 and the contract hash not be the empty code hash? That would sound like a bug to me if that ever happened

Reply:
Just explaining out my though process here. The check for `contractHash != (common.Hash{})` is useful for when the account does not yet exist in the tree. If it is an EOA and has sent a single tx then the account will exist in the tree and the `emptyCodeHash` should be returned.